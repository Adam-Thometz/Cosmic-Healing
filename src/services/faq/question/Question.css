.Question {
  position: relative;
  text-align: left;
  width: 72%;
  margin: 0 auto;
  border-bottom: 1px solid #CFC8D8;
}

.Question:last-child {
  border-bottom: none;
}

.Question-question-wrapper {
  margin: 20px 0;
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.Question-question-wrapper img {
  width: 25px;
  height: 25px;
}

.Question-question-wrapper h3 {
  margin: 0;
}

.Question-answer {
  width: 75%;
  max-height: 0;
  display: none;
  overflow: hidden;
  animation: close-answer 0.5s ease-in-out;
}

.Question-answer-showing {
  max-height: 1000px;
  display: block;
  animation: open-answer 0.5s ease-in-out;
}

@keyframes close-answer {
  from {
    max-height: 1000px;
  }
  to {
    max-height: 0;
  }
}

@keyframes open-answer {
  from {
    max-height: 0;
  }
  to {
    max-height: 1000px;
  }
}

@media (max-width: 500px) {
  .Question {
    width: 90%;
  }

  .Question-answer {
    width: 90%;
  }
}