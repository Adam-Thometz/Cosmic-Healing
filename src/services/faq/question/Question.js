import React from 'react';

import SectionHeader from '../../../_components/section-header/SectionHeader';

import './Question.css';

import plus from '../../../_media/services/expanded.png';
import minus from '../../../_media/services/collapsed.png';

const Question = ({ id, question, answer, isShowing, handleShowing }) => { 
  const handleKeyPress = e => {
    if (e.key === 'Enter' || e.key === 'Space') handleShowing();
  } 
  return (
    <div className='Question' id={id} onClick={handleShowing}>
      <div className='Question-question-wrapper'>
        <SectionHeader small text={question} />
        <img
          role='button'
          onKeyUp={handleKeyPress}
          src={isShowing ? minus : plus}
          alt={`Pick to ${isShowing ? 'collapse' : 'expand'} question`} tabIndex={0} />
      </div>
      <p className={`Question-answer${isShowing ? ' showing' : ''}`}>{answer}</p>
    </div>
  );
};

export default Question;