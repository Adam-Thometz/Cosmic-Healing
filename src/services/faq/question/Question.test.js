import React from "react";
import { render } from "@testing-library/react";
import Question from './Question';

describe('Question component', () => {
  const component = <Question
    question="What did the irritated test say?"
    answer="Don't test me"
  />;
  it('renders without crashing', () => {
    render(component);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });
});