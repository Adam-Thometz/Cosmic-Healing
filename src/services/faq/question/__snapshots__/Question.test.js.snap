// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Question"
  >
    <div
      class="Question-question-wrapper"
    >
      <h3
        class="SectionHeader"
      >
        What did the irritated test say?
      </h3>
      <img
        alt="Pick to expand question"
        src="expanded.png"
      />
    </div>
    <p
      class="Question-answer"
    >
      Don't test me
    </p>
  </div>
</DocumentFragment>
`;
